stdlib/public/core/Hasher.swift

```diff
@@ -377,7 +356,16 @@ public struct Hasher {
   /// Finalizing invalidates the hasher; additional bits cannot be combined
   /// into it, and it cannot be finalized again.
   @effects(releasenone)
-  public mutating func finalize() -> Int {
+  @usableFromInline
+  internal mutating func _finalize() -> Int {
     return Int(truncatingIfNeeded: _core.finalize())
   }
+
+  /// Finalize the hasher state and return the hash value. Finalizing consumes
+  /// the hasher, forbidding further operations.
+  @effects(releasenone)
+  public __consuming func finalize() -> Int {
```

Comment:
@jckarter, @natecook1000 A later commit changes it to read "Finalizing consumes the hasher, forbidding further operations", which is technically kind of true, but it's not the whole story. How about this?

```swift
public Hasher {
  ...
  /// Finalize the hasher state and return the hash value.
  ///
  /// Finalizing consumes the hasher: it is illegal to finalize a hasher you
  /// don't own, or to perform operations on a finalized hasher. (These
  /// may become compile-time errors in the future.)
  @effects(releasenone)
  public __consuming func finalize() -> Int {...}
}
...
public protocol Hashable {
  ...
  /// Hash the essential components of this value into the hash function
  /// represented by `hasher`, by feeding them into it using its `combine`
  /// methods.
  ///
  /// Essential components are precisely those that are compared in the type's
  /// implementation of `Equatable`.
  ///
  /// Note that `hash(into:)` doesn't own the hasher passed into it, so it must not
  /// call `finalize()` on it. Doing so may become a compile-time error in the future.
  func hash(into hasher: inout Hasher)
}

```